### PR TITLE
7.0 stock move backdating update

### PR DESCRIPTION
**Pull request by ehdem**
- Improve Code
- Generate translations
- Fix the bug where the warning message displays manytimes.

### CODE REVIEW

File: stock_move_backdating/wizard/__init__.py [hunk 1]
```diff
@@ -17,4 +17,7 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from . import stock_partial_picking
+
```
Comment (pedrobaeza) [inline stock_move_backdating/wizard/__init__.py]: Why this change? Better 1 line than 3.

File: stock_move_backdating/wizard/stock_partial_picking_view.xml [hunk 2]
```diff
@@ -8,7 +8,9 @@
             <field name="inherit_id" ref="stock.stock_partial_picking_form" />
             <field name="arch" type="xml">
                 <xpath expr="//field[@name='move_ids']/tree//field[@name='product_uom']" position="after">
-                    <field name="date_backdating" on_change="on_change_date_backdating(date_backdating)"/>
+                    <field
```
Comment (pedrobaeza) [inline stock_move_backdating/wizard/stock_partial_picking_view.xml]: I prefer to put to save one line and see more indented the attributes (more than a new tag):

```
<field name="date_backdating" 
       on_change="on_change_date_backdating(date_backdating)"/>
```

File: stock_move_backdating/view/stock_view.xml [hunk 3]
```diff
@@ -20,7 +22,9 @@
 			<field name="inherit_id" ref="stock.view_move_picking_form" />
 			<field name="arch" type="xml">
 				<xpath expr="//field[@name='date_expected']" position="after">
-					<field name="date_backdating" on_change="on_change_date_backdating(date_backdating)" />
+					<field
+            name="date_backdating"
```
Comment (pedrobaeza) [inline stock_move_backdating/view/stock_view.xml]: Bad indentation

File: stock_move_backdating/view/stock_view.xml [hunk 4]
```diff
@@ -9,7 +9,9 @@
 			<field name="inherit_id" ref="stock.view_move_form" />
 			<field name="arch" type="xml">
 				<xpath expr="//field[@name='date_expected']" position="after">
-					<field name="date_backdating" on_change="on_change_date_backdating(date_backdating)" />
+					<field
+            name="date_backdating"
```
Comment (pedrobaeza) [inline stock_move_backdating/view/stock_view.xml]: Bad indentation. Maybe it's because a mix of tabs and spaces. Please use spaces for all.

File: stock_move_backdating/wizard/__init__.py (hunk 1 above)
Comment (ehdem) [inline stock_move_backdating/wizard/__init__.py]: Because, one day we would like to add another model, It will be more simple

Comment (pedrobaeza) [inline stock_move_backdating/wizard/__init__.py]: Well, I prefer to put in all lines `from . import ...`, but it's not so important.

File: stock_move_backdating/stock.py [hunk 5]
```diff
@@ -81,10 +81,10 @@ def on_change_date_backdating(self, cr, uid, ids, date_backdating,
         if (dt > NOW):
             warning = {'title': _('Error!'), 'message': _(
                 'You can not process an actual movement date in the future.')}
-            values = {
-                'date_backdating': NOW.strftime(DEFAULT_SERVER_DATETIME_FORMAT)
-            }
```
Comment (llacroix) [inline stock_move_backdating/stock.py:86]: Doesn't it actually break the functionality of the module? As the values aren't returned the `date_backdating` will remain unchanged and it will allow the user to set a date in the future. Yet the module tells us that we cannot do that... that seems strange.

If the warning is sent multiple times it might be caused by simultaneous calls to the `on_change` event but if it was being called multiple times, it should change the date which seems strange.

File: stock_move_backdating/wizard/stock_partial_picking_view.xml [hunk 6]
```diff
@@ -1,17 +1,16 @@
...
-            <field name="model">stock.partial.picking</field>
-            <field name="inherit_id" ref="stock.stock_partial_picking_form" />
-            <field name="arch" type="xml">
-                <xpath expr="//field[@name='move_ids']/tree//field[@name='product_uom']" position="after">
-                    <field name="date_backdating" on_change="on_change_date_backdating(date_backdating)"/>
-                </xpath>
-            </field>
-        </record>
-
+      <record id="stock_partial_picking_form" model="ir.ui.view">
```
Comment (llacroix) [inline stock_move_backdating/wizard/stock_partial_picking_view.xml]: You changed the indentation to 2 it was 4 but some elements are still indented to 4 `<openerp><data>`.

File: stock_move_backdating/view/stock_view.xml (hunk 3 above)
Comment (llacroix) [inline stock_move_backdating/view/stock_view.xml]: @pedrobaeza It appears that the file was indented with tabs.

File: stock_move_backdating/stock.py (hunk 5 above)
Comment (ehdem) [inline stock_move_backdating/stock.py:86]: @llacroix : It doesn't break the functionality. The bug for multiple popup message is only for date field. After popup message you change the date if you want.

Comment (llacroix) [inline stock_move_backdating/stock.py:86]: I understand but the `value` is setting the date that is in the future back to the `today` date to prevent setting date in the future.
After your change, it might be possible to set a date in the future as the warning isn't preventing changes. It's just a warning.

Comment (ehdem) [inline stock_move_backdating/stock.py:86]: With my changes, we can add a constraint (for date_backdating field) when creating or writing

File: stock_move_backdating/__openerp__.py [hunk 7]
```diff
@@ -19,22 +19,34 @@
...
+
+Contributors
+------------
+* Marco Dieckhoff
+* BREMSKERL
+* Agile Business Group
+* El Hadji Dem (elhadji.dem@savoirfairelinux.com)
+* David Dufresne (david.dufresne@savoirfairelinux.com)
+    """,
+    'author': 'Bremskerl',
```
Comment (gurneyalex) [inline stock_move_backdating/__openerp__.py]: could you add "Odoo Community Association (OCA)" to the authors, so that the module shows up under the association on app.odoo.com?

**Comment by gurneyalex** [comment]
with the small requested change, :+1:

**Comment by ehdem** [comment]
Can't access the log from https://runbot.odoo-community.org/runbot/build/2926249

**Comment by jgrandguillaume** [comment]
I see that in the log:

I ask to rebuild, seems a trouble with dependencies not related to this module. A part from this trouble, I'm :+1: 

---

Failed to initialize database `2947093-54-cac13b-all`.
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2947093-54-cac13b/openerp/service/server.py", line 919, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2947093-54-cac13b/openerp/modules/registry.py", line 370, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2947093-54-cac13b/openerp/modules/loading.py", line 320, in load_modules
    modobj.button_install(cr, SUPERUSER_ID, ids)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2947093-54-cac13b/openerp/api.py", line 241, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2947093-54-cac13b/openerp/addons/base/module/module.py", line 417, in button_install
    self.state_update(cr, uid, ids, 'to install', ['uninstalled'], context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2947093-54-cac13b/openerp/api.py", line 241, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2947093-54-cac13b/openerp/api.py", line 363, in old_api
    result = method(recs, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2947093-54-cac13b/openerp/addons/base/module/module.py", line 396, in state_update
    raise orm.except_orm(_('Error'), _("You try to install module '%s' that depends on module '%s'.\nBut the latter module is not available in your system.") % (module.name, dep.name,))
except_orm: ('Error', u"You try to install module 'product_customer_code_picking' that depends on module 'product_customer_code'.\nBut the latter module is not available in your system.")

---

File: stock_move_backdating/__openerp__.py [hunk 8]
```diff
@@ -19,22 +19,34 @@
...
+
+Contributors
+------------
+* Marco Dieckhoff
+* BREMSKERL
+* Agile Business Group
+* El Hadji Dem (elhadji.dem@savoirfairelinux.com)
+* David Dufresne (david.dufresne@savoirfairelinux.com)
+    """,
+    'author': 'Bremskerl, Odoo Community Association (OCA)',
```
Comment (tafaRU) [inline stock_move_backdating/__openerp__.py]: What's the reason to have removed `Marco Dieckhoff, Agile Business Group` from author?

Comment (max3903) [inline stock_move_backdating/__openerp__.py]: None, fixed.

**Comment by max3903** [comment]
:+1:
